types/lib/model.d.ts

```diff
@@ -3081,24 +3080,33 @@ export type CreationOptional<T> = T extends Array<infer Val>
 export type InferCreationAttributes<
   M extends Model,
   Options extends InferAttributesOptions<keyof M | never | ''> = { omit: never }
-> = {
-  [Key in keyof M as
-    M[Key] extends AnyFunction ? never
-    : { [NonAttributeBrand]: true } extends M[Key] ? never
-      // array special case
-    : M[Key] extends NonAttributeBrandedArray<any> ? never
-    : Key extends keyof Model ? never
-    // check 'omit' option is provided
-    : Options['omit'] extends string ? (Key extends Options['omit'] ? never : Key)
-    : Key
-     // it is normal that the brand extends the type.
-     // We're checking if it's in the type.
-  ]: { [CreationAttributeBrand]: true } extends M[Key] ? (M[Key] | undefined)
-    // array special case
-    : CreationAttributeBrandedArray<any> extends M[Key] ? (M[Key] | undefined)
+  > = {
+  [Key in keyof M as InternalInferAttributeKeysFromFields<M, Key, Options>]: IsBranded<M[Key], typeof CreationAttributeBrand> extends true
+    ? (M[Key] | undefined)
     : M[Key]
 };
 
+/**
+ * @private
+ *
+ * Internal type used by {@link InferCreationAttributes} and {@link InferAttributes} to exclude
+ * attributes that are:
+ * - functions
+ * - branded using {@link NonAttribute}
+ * - inherited from {@link Model}
+ * - Excluded manually using {@link InferAttributesOptions#omit}
+ */
+type InternalInferAttributeKeysFromFields<M extends Model, Key extends keyof M, Options extends InferAttributesOptions<keyof M | never | ''>> =
+  // functions are always excluded
+  M[Key] extends AnyFunction ? never
+  // fields inherited from Model are all excluded
+  : Key extends keyof Model ? never
+  // fields branded with NonAttribute are excluded
+  : IsBranded<M[Key], typeof NonAttributeBrand> extends true ? never
+  // check 'omit' option is provided & exclude those listed in it
+  : Options['omit'] extends string ? (Key extends Options['omit'] ? never : Key)
```

Comment:
It's a type but meant to be string or union of strings
It's used as an escape hatch if it turns out there is an issue with branded types and the user needs another way to omit types, instead of `NonAttribute<T>`.

It's used like this:

```typescript
class User extends Model<InferAttributes<User, { omit: 'projects' | 'groups' }>> {
  id: number;
  name: string;
  projects?: Project[];
  groups?: Group[];
}
```

Reply:
Just in case, it's documented here :) https://github.com/sequelize/sequelize/blob/v6/docs/manual/other-topics/typescript.md